### PR TITLE
DEV: Adds a rake task to convert the reactions

### PR DESCRIPTION
**Pull request by Ahmedgagan**
@jjaffeux 
This rake task convert's the reaction to another OR drops the reaction.
This was discussed here https://meta.discourse.org/t/discourse-reactions-beyond-likes/183261/106?u=ahmed_gagan

### CODE REVIEW

File: lib/tasks/reactions.rake [hunk 1]
```diff
@@ -50,3 +50,48 @@ task "reactions:generate", [:post_id, :reactions_count, :reaction] => [:environm
...
+
+  reaction = DiscourseReactions::Reaction.find_by(reaction_value: old_reaction)
+
+  if !reaction
+    raise "Please provide an existing reaction in 'old_reaction' parameter and try again!"
+  end
+
+  puts "Converting reactions"
+  reaction.reaction_users.each do |reaction_user|
+    post = Post.find_by(id: reaction_user.post_id)
```
Comment (jjaffeux) [inline lib/tasks/reactions.rake]: cant you join users ands posts in reaction users and use it fo lookup post/user ?

File: lib/tasks/reactions.rake [hunk 2]
```diff
@@ -50,3 +50,48 @@ task "reactions:generate", [:post_id, :reactions_count, :reaction] => [:environm
...
+
+    DiscourseReactions::ReactionManager.new(
+      reaction_value: new_reaction ? new_reaction : old_reaction,
+      user: user,
+      guardian: Guardian.new(user),
+      post: post
+    ).toggle!
+  end
+
+  SiteSetting.post_undo_action_window_mins = POST_UNDO_ACTION_WINDOW_MINS
```
Comment (jjaffeux) [inline lib/tasks/reactions.rake]: I would prefer if all of this was in a begin/ensure block, so we can be certain whatever happen that we are restoring the setting.

File: lib/tasks/reactions.rake (hunk 1 above)
Comment (Ahmedgagan) [reply lib/tasks/reactions.rake]: I think we don't need the join, I used the associations here in the latest commit

File: lib/tasks/reactions.rake (hunk 2 above)
Comment (Ahmedgagan) [reply lib/tasks/reactions.rake]: Indeed, added a commit with begin/ensure block

**Comment by Ahmedgagan** [comment]
@jjaffeux i have added commits with suggested changes. Can you take a look?

**Pull request closed without merging**